### ISSUE TITLE
Change the folder where public upload chunks are stored

### ISSUE THREAD

**Issue by JMarcosHP** [nextcloud/all-in-one#5496]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
Currently public file uploads are temporarily stored in /tmp and then moved to the actual location in the nextcloud data directory.

This can be a problem for users who do not have a dedicated partition with fixed size for /tmp or users who use Nextcloud-AIO in docker as it can completely fill up the partition or location where they have the docker installation files.

**Describe the solution you'd like**

It is known that for nextcloud 31 public file uploads will be [migrated](https://github.com/nextcloud/server/issues/4109#issuecomment-2232763179) to vue in the application files and will support chunking, so why not take advantage and change the way they are stored temporarily, for example in the data directory as well as uploads are stored for registered users (`data/user/uploads`) can be `/data/public/uploads` in this case.

**Additional context**

 ([See](https://github.com/nextcloud/all-in-one/discussions/5356))

**Comment by susnux** [comment]
I transferred this from server to aio as this is purely a configuration issue.
For PHP tempfiles are stored in `upload_tmp_dir` and for the temporary files you can use the Nextcloud [tempdirectory](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/config_sample_php_parameters.html#tempdirectory) setting.

So this is just configuration of AiO

**Comment by ErikSteiner** [comment]
> [@susnux](https://github.com/susnux) Steps to reproduce:
> 
>     1. Make your custom temp directory on a host disk (SSD recommended), then add the mountpoint on `NEXTCLOUD_MOUNT`:
> ...

Do I understand correctly that the adjustment of the temporary directory involves two parts that have nothing to do with each other? As far as I understand it, can the temporary directory of PHP be in a different location than the temporary directory of Nextcloud?

**Comment by JMarcosHP** [comment]
> > [@susnux](https://github.com/susnux) Steps to reproduce:
> > 
> >     1. Make your custom temp directory on a host disk (SSD recommended), then add the mountpoint on `NEXTCLOUD_MOUNT`:
> > ...
> 
> Do I understand correctly that the adjustment of the temporary directory involves two parts that have nothing to do with each other? As far as I understand it, can the temporary directory of PHP be in a different location than the temporary directory of Nextcloud?
> 
> As far as I can see it, I have just an issue with the temporary files of PHP then, because those phpXXXX folders are filling up my container and the the docker host.

Yes sir, we can use the temp directory of nextcloud but this directive is ignored when you upload files, since PHP and PHP-FPM have their own ENV variables for the tmp directory is better to point all to a fixed mountpoint where Nextcloud and PHP stores all his files to avoid issues.

**Comment by ErikSteiner** [comment]
OK, but is there any known interaction between PHP-FPM and Nextcloud, that would require a path, that match for both?

**Comment by JMarcosHP** [comment]
I tested, and yes.
The Nextcloud docker image uses PHP-FPM, some apps like memories save temporary files, for example when you download geocode database.
